### PR TITLE
Docs: Convert Extending and Answers chapters

### PR DESCRIPTION
**Pull request by cmills1095**
This pull request, together with the others I've submitted, should wrap up everything in the manual that isn't a cookbook, benchmark, or parameter file.

### CODE REVIEW

**Comment by cmills1095** [comment]
I have a feeling I'm going to have to change some things once https://github.com/geodynamics/aspect/pull/4745 is approved. Let me know!

**Comment by gassmoeller** [comment]
#4745 is merged, do you want me to review this now, or adjust this PR first?

**Comment by cmills1095** [comment]
> #4745 is merged, do you want me to review this now, or adjust this PR first?

Ready for review

File: doc/sphinx/user/extending/idea-of-plugins.md [hunk 1]
```diff
@@ -4,36 +4,38 @@
...
-</figure>
+```{figure-md} fig:plugins
+<img src="../../../manual/plugin_graph.*" alt="The graph of all current plugins of ASPECT. The yellow octagon and square represent the `Simulator` and [SimulatorAccess](doc/doxygen/classaspect_1_1SimulatorAccess.html) classes. The green boxes are interface classes for everything that can be changed by plugins. Blue circles correspond to plugins that implement particular behavior. The graph is of course too large to allow reading individual plugin names (unless you zoom far into the page), but is intended to illustrate the architecture of ASPECT."  width="95%"/>
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/idea-of-plugins.md:19]: do you want to move this over into the new place `images/plugin_graph` already?

File: doc/sphinx/user/extending/idea-of-plugins.md [hunk 2]
```diff
@@ -83,9 +85,15 @@ The way this is achieved is through the following two steps:
     Here, the first argument to the macro is the name of the class. The second
     is the name by which this model can be selected in the parameter file. And
     the third one is a documentation string that describes the purpose of the
-    class (see, for example, {ref}`sec:3.55][] for an example of how
+    class (see, for example, {ref}`sec:3.55` for an example of how
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/idea-of-plugins.md:88]: this is probably a different reference now? I think it is:
```suggestion
    class (see, for example, {ref}`parameters:Gravity_20model` for an example of how
```

File: doc/sphinx/user/extending/idea-of-plugins.md [hunk 3]
```diff
@@ -83,9 +85,15 @@ The way this is achieved is through the following two steps:
...
     existing models describe themselves).
 
+    :::{admonition} TODO
+    :class: error
+
+    {ref}`sec:3.55` refers to a parameter file; these have not yet been set up
+    :::
+
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/idea-of-plugins.md:96]:
```suggestion
```

File: doc/sphinx/user/extending/index.md [hunk 4]
```diff
@@ -134,9 +133,13 @@ which there is an extensive amount of documentation:
     lists at <https://www.dealii.org/mail.html> are also of general help.
 
 -   A general overview of deal.II is also
-    provided in the paper (Bangerth, Hartmann, and Kanschat 2007).
+    provided in the paper {cite:t}`bangerth:etal:2007`.
 
-As described in {ref}`sec:debug-mode`7] you should always compile
+:::{admonition} TODO
+<https://www.dealii.org/developer/doxygen/tutorial/index.html> no longer seems to go anywhere.
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/index.md:139]: Replace with https://dealii.org/developer/doxygen/deal.II/Tutorial.html

File: doc/sphinx/user/extending/plugin-types/mesh-refinement.md [hunk 5]
```diff
@@ -16,19 +16,23 @@ automatic refinement criteria do not always refine these interfaces as well as
...
 
 To alleviate this, ASPECT has plugins for mesh
-refinement. Through the parameters in {ref}`sec:3.129][], one can select
+refinement. Through the parameters in {ref}`sec:3.129`, one can select
 when to refine but also which refinement criteria should be used and how they
 should be combined if multiple refinement criteria are selected. Furthermore,
 through the usual plugin mechanism, one can extend the list of available mesh
 refinement criteria (see the parameter "Strategy" in
-{ref}`sec:3.129][]). Each such plugin is responsible for producing a
+{ref}`sec:3.129`). Each such plugin is responsible for producing a
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/plugin-types/mesh-refinement.md:24]:
```suggestion
{ref}`parameters:Mesh_20refinement/Strategy`). Each such plugin is responsible for producing a
```

File: doc/sphinx/user/extending/plugin-types/mesh-refinement.md [hunk 6]
```diff
@@ -16,19 +16,23 @@ automatic refinement criteria do not always refine these interfaces as well as
...
 vector of values (one per active cell on the current processor, though only
 those values for cells that the current processor owns are used) with an
 indicator of how badly this cell needs to be refined: large values mean that
 the cell should be refined, small values that the cell may be coarsened away.
 
+:::{admonition} TODO
+`sec:3.129` references are to parameter files which are not yet set up.
+:::
+
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/plugin-types/mesh-refinement.md:33]:
```suggestion


```

File: doc/sphinx/user/extending/plugin-types/postprocessors.md [hunk 7]
```diff
@@ -106,13 +107,13 @@ Primarily, this difficulty results from two facts:
...
-    [aspect::Postprocess::Interface][] base class, postprocessors also derive
-    from the [SimulatorAccess][aspect::SimulatorAccess class] class that has a
+    `aspect::Postprocess::Interface` base class, postprocessors also derive
+    from the `SimulatorAccess` class that has a
     number of member functions postprocessors can call to obtain read-only
     access to some of the information stored in the main class of
-    ASPECT. See [the documentation of this
-    class][aspect::SimulatorAccess class] to see what kind of information is
-    available to postprocessors. See also {ref}`sec:1.1][] for more
+    ASPECT. See the documentation of the `aspect::SimulatorAccess class` to see
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/plugin-types/postprocessors.md:114]:
```suggestion
    ASPECT. See the documentation of the [aspect::SimulatorAccess class](https://aspect.geodynamics.org/doc/doxygen/classaspect_1_1SimulatorAccess.html) to see
```

File: doc/sphinx/user/extending/plugin-types/terminate-criteria.md [hunk 8]
```diff
@@ -7,10 +7,14 @@ simulation when it has reached a steady state (or, rather, some criterion
 determines that it is close enough to steady state), or by an external action
 such as placing a specially named file in the output directory. The criteria
 determining termination of a simulation are all implemented in plugins. The
-parameters describing these criteria are listed in {ref}`sec:3.195][].
+parameters describing these criteria are listed in {ref}`sec:3.195`.
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/plugin-types/terminate-criteria.md:10]:
```suggestion
parameters describing these criteria are listed in {ref}`parameters:Termination_20criteria`.
```

File: doc/sphinx/user/extending/plugin-types/terminate-criteria.md [hunk 9]
```diff
@@ -7,10 +7,14 @@ simulation when it has reached a steady state (or, rather, some criterion
...
+
+:::{admonition} TODO
+sec:3.195 refers to a parameter file which is not set up yet.
+:::
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/plugin-types/terminate-criteria.md:14]:
```suggestion
```

File: doc/sphinx/user/extending/plugin-types/vis-postprocessors.md [hunk 10]
```diff
@@ -18,7 +18,11 @@ outputting *all* of them would yield very large output files, and would
 furthermore not scale very well as the list continues to grow. Consequently,
 as with the postprocessors described in the previous section, what *can* be
 computed is implemented in a number of plugins and what *is* computed is
-selected in the input parameter file (see {ref}`sec:3.167][]).
+selected in the input parameter file (see {ref}`sec:3.167`).
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/plugin-types/vis-postprocessors.md:21]:
```suggestion
selected in the input parameter file (see {ref}`parameters:Postprocess/Visualization`).
```

File: doc/sphinx/user/extending/plugin-types/vis-postprocessors.md [hunk 11]
```diff
@@ -18,7 +18,11 @@ outputting *all* of them would yield very large output files, and would
...
+
+:::{note}
+Reference to sec;3.167 is a reference to a parameter file which is not currently set up.
+:::
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/plugin-types/vis-postprocessors.md:25]:
```suggestion
```

File: doc/sphinx/user/extending/plugin-types/vis-postprocessors.md [hunk 12]
```diff
@@ -27,9 +31,9 @@ by convention be in namespace
 `aspect::Postprocess::VisualizationPostprocessors`, and is registered using a
 macro, here called `ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR`. Like the
 postprocessor plugins, visualization postprocessors can derive from class
-[aspect::Postprocess::SimulatorAccess][] if they need to know specifics of the
+`aspect::Postprocess::SimulatorAccess` if they need to know specifics of the
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/plugin-types/vis-postprocessors.md:34]: This was wrong before.
```suggestion
`aspect::SimulatorAccess` if they need to know specifics of the
```

File: doc/sphinx/user/extending/write-a-plugin.md [hunk 13]
```diff
@@ -58,7 +59,7 @@ to do:
         shared library at startup so that the plugin becomes available at run
         time and can be selected from the input parameter file. This is done
         using the `Additional shared libraries` parameter in the input file,
-        see {ref}`sec:3.1][]. This approach has the upside that you can
+        see {ref}`sec:3.1`. This approach has the upside that you can
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/write-a-plugin.md:62]:
```suggestion
        see {ref}`parameters:Additional_20shared_20libraries`. This approach has the upside that you can
```

File: doc/sphinx/user/extending/write-a-plugin.md [hunk 14]
```diff
@@ -82,26 +84,43 @@ to do:
...
+:::{note}
+If you choose to compile your plugins into a shared library yourself, you will need to
+recompile them every time you upgrade your ASPECT installation since we do not guarantee
+that the ASPECT application binary interface (ABI) will remain stable, even if it may not be
+necessary to actually change anything in the *implementation* of your plugin
+:::
+
+:::{admonition} TODO
+{ref}`sec:3.1` refers to a parameter file, which is not set up yet.
+:::
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/write-a-plugin.md:126]:
```suggestion
```

**Review by gassmoeller — COMMENTED** [review]
Very nice. Just a few reference comments.

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**